lib/spack/spack/config.py

```diff
@@ -773,6 +782,7 @@ def _add_command_line_scopes(cfg, command_line_scopes):
         name = "cmd_scope_%d" % i
         cfg.push_scope(ImmutableConfigScope(name, path))
         _add_platform_scope(cfg, ImmutableConfigScope, name, path)
+        _add_os_scope(cfg, ImmutableConfigScope, name, path)
```

Comment:
I think you'll also want to do this in the other location where `_add_platform_scope` is called, in `_config` (otherwise only config directories added with `spack -C ...` will include OS scopes, and not e.g. `~/.spack/`.

Reply:
I thought I'd done that... Thanks.   Pushed.